### DISCUSSION TITLE
Flux installation

### DISCUSSION THREAD

**Discussion by kuratowskiy** [fluxcd/flux2#1217] [Q&A]
According to the guide, I created a kustomization inside my flux-system folder, that includes the gotk-sync.yaml, that captures my fleet repository information. 

When I start my cluster, I apply the kustomization, but it complains about the flux repository resources, even though the custom resource definitions should have already been applied? 

If after I apply individually the gotk-sync.yaml, it works. 

Why should I not be able to just apply the two at once?

**Comment by stefanprodan** [comment]
> Why should I not be able to just apply the two at once?

This is how Kubernetes works, CRDs must be applied before CRs.

**Reply by kuratowskiy** [reply]
I think the guide might be revised then? It suggests to create a kustomization including the dependent manifest.


Also parenthetically Helm reasonably applies CRDs before resources.

**Reply by stefanprodan** [reply]
The guide tells you to apply gotk-components.yaml before the kustomization
